### PR TITLE
Cleanup module dependencies

### PR DESCRIPTION
**Pull request by larskuhtz**
This cleans up the module dependency graph by moving some types into modules on their own.

This is preparatory work for some upcoming feature PRs.

### CODE REVIEW

File: src/Chainweb/Payload.hs [hunk 1]
```diff
@@ -488,12 +483,22 @@ coinbaseOutputFromText t = either (throwM . TextFormatException . sshow) return
...
+        ]
+    , "reqKey" .= ("DldRwCblQ7Loqy6wYJnaodHl30d3j3eH-qtFzfEv46g" :: String)
+        -- this is the unique hash value define in @Pact.Types.Hash.initialHash@
+    , "logs" .= Null
+    , "metaData" .= Null
+    , "continuation" .= Null
+    , "txId" .= Null
+    ]
+{-# NOINLINE noCoinbaseOutput #-}
+
```
Comment (fosskers) [inline src/Chainweb/Payload.hs]: Why did this change?

Comment (larskuhtz) [reply src/Chainweb/Payload.hs]: `Chainweb.Payload` was supposed to not depend on pact internals. `noCoinBaseOutput` is a constant that will never change. So it's fine to directly provide the content and avoiding  using constructors from pact types. The PR adds a unit test to ensure that both are in sync.

This simplifies the module graph a lot.

Comment (larskuhtz) [reply src/Chainweb/Payload.hs]: Oh, and this value is only used in test-version of chainweb. It's not used in production.

File: src/Chainweb/BlockHeight.hs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+encodeBlockHeight (BlockHeight h) = putWord64le h
+
+decodeBlockHeight :: MonadGet m => m BlockHeight
+decodeBlockHeight = BlockHeight <$> getWord64le
+
+encodeBlockHeightBe :: MonadPut m => BlockHeight -> m ()
+encodeBlockHeightBe (BlockHeight r) = putWord64be r
+
+decodeBlockHeightBe :: MonadGet m => m BlockHeight
+decodeBlockHeightBe = BlockHeight <$> getWord64be
```
Comment (emilypi) [inline src/Chainweb/BlockHeight.hs]: can we a get a note describing under what circumstances we need big and little endian? Also, while we're shuffling things around, the convention would be to disambiguate with `encodeBlockHeight` with `encodeBlockHeightLe` to contrast the other encoding (ditto for decoding)

File: src/Chainweb/Payload.hs (hunk 1 above)
Comment (emilypi) [reply src/Chainweb/Payload.hs]: Yes, thank you. The pact dependency here should definitely go away.

File: src/Chainweb/BlockHeight.hs (hunk 2 above)
Comment (larskuhtz) [reply src/Chainweb/BlockHeight.hs]: Good point. I think, we don't use bigendian anywhere any more. I think we can just remove this function.

**Pull request merged**